### ISSUE TITLE
The line of the method that threw an exception in the stack trace is missing.

### ISSUE THREAD

**Issue by mayuki** [Cysharp/UniTask#105]
The line of the method that threw an exception in the stack trace is missing.
For example, when the following code is executed, the line for the `Throw` and `MethodFooAsync`, `MethodBarAsync` method are missing.

## Reproducible code
```csharp
public class UniTaskTestScene : MonoBehaviour
{
    public Button ButtonTest;

    void Start()
    {
        ButtonTest.OnClickAsAsyncEnumerable()
            .Subscribe(async _ =>
            {
                try
                {
                    await new Foo().MethodFooAsync();
                }
                catch (Exception e)
                {
                    Debug.Log(e.StackTrace);
                }
            }, this.GetCancellationTokenOnDestroy());
    }
}

class Foo
{
    public async UniTask MethodFooAsync()
    {
        await MethodBarAsync();
    }

    private async UniTask MethodBarAsync()
    {
        Throw();
    }

    private void Throw()
    {
        throw new Exception();
    }
}
```

### Stack trace
```
  at Cysharp.Threading.Tasks.UniTask+ExceptionResultSource.GetResult (System.Int16 token) [0x00001] in C:\Path\To\Project\Library\PackageCache\com.cysharp.unitask@c49f1ed028\Runtime\UniTask.Factory.cs:188 
  at Cysharp.Threading.Tasks.UniTask+Awaiter.GetResult () [0x00015] in C:\Path\To\Project\Library\PackageCache\com.cysharp.unitask@c49f1ed028\Runtime\UniTask.cs:307 
  at UniTaskTestScene+<>c+<<Start>b__1_0>d.MoveNext () [0x0002b] in C:\Path\To\Project\Assets\Scripts\UniTaskTestScene.cs:20 
```

## Environment
- Unity 2019.4.f1 (Windows Editor)
- UniTask 2.0.22

**Comment by neuecc** [comment]
Thanks.

This is caused by 

`AsyncUniTaskMethodBuilder.Task` -> `return UniTask.FromException(ex);`
-> `return new UniTask(new ExceptionResultSource(ex), 0);` -> `public void GetResult(short token) => throw exception`

It has completely lost exception info.
This is critical, I should fix immediately.

**Comment by neuecc** [comment]
fixed stacktrace

```
  at Foo.Throw () [0x00001] in C:\Users\neuecc\Source\Repos\UniTask\src\UniTask\Assets\Scenes\ExceptionExamples.cs:44 
  at Foo+<MethodBarAsync>d__1.MoveNext () [0x00008] in C:\Users\neuecc\Source\Repos\UniTask\src\UniTask\Assets\Scenes\ExceptionExamples.cs:39 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <437ba245d8404784b9fbab9b439ac908>:0 
  at Cysharp.Threading.Tasks.UniTask+ExceptionResultSource.GetResult (System.Int16 token) [0x00001] in C:\Users\neuecc\Source\Repos\UniTask\src\UniTask\Assets\Plugins\UniTask\Runtime\UniTask.Factory.cs:189 
  at Cysharp.Threading.Tasks.UniTask+Awaiter.GetResult () [0x00015] in C:\Users\neuecc\Source\Repos\UniTask\src\UniTask\Assets\Plugins\UniTask\Runtime\UniTask.cs:307 
  at Foo+<MethodFooAsync>d__0.MoveNext () [0x00023] in C:\Users\neuecc\Source\Repos\UniTask\src\UniTask\Assets\Scenes\ExceptionExamples.cs:34 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <437ba245d8404784b9fbab9b439ac908>:0 
  at Cysharp.Threading.Tasks.UniTask+ExceptionResultSource.GetResult (System.Int16 token) [0x00001] in C:\Users\neuecc\Source\Repos\UniTask\src\UniTask\Assets\Plugins\UniTask\Runtime\UniTask.Factory.cs:189 
  at Cysharp.Threading.Tasks.UniTask+Awaiter.GetResult () [0x00015] in C:\Users\neuecc\Source\Repos\UniTask\src\UniTask\Assets\Plugins\UniTask\Runtime\UniTask.cs:307 
  at ExceptionExamples+<>c+<<Start>b__1_0>d.MoveNext () [0x0002b] in C:\Users\neuecc\Source\Repos\UniTask\src\UniTask\Assets\Scenes\ExceptionExamples.cs:20 
```

**Comment by neuecc** [comment]
It is caused when task is `sync` completed.

**Comment by neuecc** [comment]
fixed at `2.0.23`.
